### PR TITLE
When img src is missing, avoid break the app

### PR DESCRIPTION
**Pull request by flaviohenriquecbc**
- correction to avoid break the app when the src is missing

### CODE REVIEW

File: htmlToElement.js [hunk 1]
```diff
@@ -1,13 +1,14 @@
 import React from 'react';
-import {StyleSheet, Text} from 'react-native';
+import {StyleSheet, Text, Dimensions, View} from 'react-native';
```
Comment (richchurcher) [inline htmlToElement.js]: `View` is unused (except in comments). Use of `Dimensions` is interesting, but outside the scope of your PR.

File: htmlToElement.js [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
 import htmlparser from 'htmlparser2-without-node-native';
 import entities from 'entities';
 
 import AutoSizedImage from './AutoSizedImage';
+import colors from '../../app/style/color.style';
 
 const defaultOpts = {
   lineBreak: '\n',
-  paragraphBreak: '\n\n',
+  paragraphBreak: '\n',
```
Comment (richchurcher) [inline htmlToElement.js]: I agree the paragraph break stuff is tricky, but again it's not really what you described in the PR title :smile: ... try to keep the changes constrained to what you're trying to fix.

File: htmlToElement.js [hunk 3]
```diff
@@ -1,13 +1,14 @@
 import React from 'react';
-import {StyleSheet, Text} from 'react-native';
+import {StyleSheet, Text, Dimensions, View} from 'react-native';
 import htmlparser from 'htmlparser2-without-node-native';
 import entities from 'entities';
 
 import AutoSizedImage from './AutoSizedImage';
+import colors from '../../app/style/color.style';
```
Comment (richchurcher) [inline htmlToElement.js]: This file doesn't exist, so I can only assume it's part of your own project?

File: htmlToElement.js [hunk 4]
```diff
@@ -16,24 +17,39 @@ const defaultOpts = {
...
     parseInt(props.attribs['height'], 10) ||
     parseInt(props.attribs['data-height'], 10) ||
     0;
 
+  const { width } = Dimensions.get('window');
+  const resizeMode = null;
+
+  if (!widthAtt) {
+    widthAtt = (width - 20);
+    resizeMode: 'contain'
```
Comment (richchurcher) [inline htmlToElement.js]: I feel like some of this stuff would be better off in `AutoSizedImage`, which after all is supposed to take care of its own sizing :smile:

File: htmlToElement.js [hunk 5]
```diff
@@ -16,24 +17,39 @@ const defaultOpts = {
...
-    width,
+    width: widthAtt,
     height,
   };
-  return <AutoSizedImage source={source} style={imgStyle} />;
+
+  if (props.attribs.src) {
+    return <AutoSizedImage source={source} style={imgStyle} />;
+  } else {
+    return <Text>-</Text>
```
Comment (richchurcher) [inline htmlToElement.js]: I guess so. It feels weird returning a `Text` where the code expects an `Image`. I wonder if it's worth including a tiny static 'x' image to indicate broken images...

File: htmlToElement.js [hunk 6]
```diff
@@ -42,6 +58,18 @@ export default function htmlToElement(rawHtml, customOpts = {}, done) {
     ...customOpts,
   };
 
+  function cleanLink(link) {
+    if (link) {
```
Comment (richchurcher) [inline htmlToElement.js]: Again, please restrict your PR to the thing you're trying to fix.

File: htmlToElement.js [hunk 7]
```diff
@@ -134,20 +173,30 @@ export default function htmlToElement(rawHtml, customOpts = {}, done) {
             </TextComponent>);
           } else if (parent.name === 'ul') {
             listItemPrefix = (<TextComponent style={[defaultStyle, customStyle]}>
-              {opts.bullet}
+              {/* <View style={inheritedStyle({name: 'bullet'})} /> */}
```
Comment (richchurcher) [inline htmlToElement.js]: Please don't submit commented out code :smile:

File: htmlToElement.js (hunk 1 above)
Comment (flaviohenriquecbc) [reply htmlToElement.js]: ops, sorry! i made a mess on this PR. I was testing something and shared my master (with a lot of tests and bad code just to see how it works.) Soon I will send the proper fixes. Ignore this PR complete. And thank you for your review.

Comment (richchurcher) [reply htmlToElement.js]: @flaviohenriquecbc That's ok, thanks for thinking of us! Very happy to take a look at your next offering :smile:

**Pull request closed without merging**